source/SelectableGridComponent/SelectableGridComponent.js

```diff
@@ -0,0 +1,123 @@
+import { Component, PropTypes } from 'react'
+
+export default class SelectableGridComponent extends Component {
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      selectedRows: props.selectedRows
+    }
+
+    this.handleCellStyle = this.handleCellStyle.bind(this)
+    this.handleOnSelect = this.handleOnSelect.bind(this)
+  }
+
+  static propTypes= {
+    /** Optional option to set if the a FlexTable allows multiple selection*/
+    allowsMultipleSelection: PropTypes.bool,
+
+    /** Children of the component, it only support the FlexTable so far*/
+    children: PropTypes.func.isRequired,
+
+    /**
+    * Optional callback when a row is selected
+    * ({ index: number }): void
+    */
+    onRowSelect: PropTypes.func,
+
+    /**
+    * Optional callback when a row is deselected
+    * ({ index: number }): void
+    */
+    onRowDeselect: PropTypes.func,
+
+    /**
+     * Callback invoked when a user clicks on a table row.
+     * ({ index: number }): void
+     */
+    onRowClick: PropTypes.func,
```

Comment:
I don't think you need to expose a redundant `onRowClick` callback here. Same goes for `rowWrapperStyle`. If users want to _also_ listen for clicks separately from select/deselect events- (which I'm not sure of)- or provide additional custom style- they could do so this:

```js
<SelectableGridComponent {...selectableGridComponentProps}>
  {({ rowWrapperStyle, onRowClick }) => (
    <FlexTable
      onRowClick={() => {
        usersOnClickHandler()
        onRowClick()
      }}
      rowWrapperStyle={{
        ...usersCustomStyle,
        ...rowWrapperStyle
      }}
      {...flexTableProps}
    >
      {...flexColumns}
    </FlexTable>
  )}
</SelectableGridComponent>
```